bin/taiyi-challenger/src/preconf_request_data.rs

```
@@ -0,0 +1,53 @@
+use std::{any::type_name, fmt::Debug};
+
+use bincode::{decode_from_slice, encode_to_vec, Decode, Encode};
+use redb::{TypeName, Value};
+
+#[derive(Debug, Clone, Decode, Encode, PartialEq)]
+pub struct PreconfRequestData {
+    pub preconf_type: u8,                  // 0: Type A, 1: Type B
+    pub preconf_request: String,           // Serde json string
+    pub preconf_request_signature: String, // Hex encoded signature
+}
+
+/// Wrapper type to handle values using bincode serialization
+#[derive(Debug)]
+pub struct Bincode<T>(pub T);
+
+impl<T> Value for Bincode<T>
+where
+    T: Debug + Encode + Decode<()>,
+{
+    type SelfType<'a>
+        = T
+    where
+        Self: 'a;
+
+    type AsBytes<'a>
+        = Vec<u8>
+    where
+        Self: 'a;
+
+    fn fixed_width() -> Option<usize> {
+        None
+    }
+
+    fn from_bytes<'a>(data: &'a [u8]) -> Self::SelfType<'a>
+    where
+        Self: 'a,
+    {
+        decode_from_slice(data, bincode::config::standard()).expect("Failed to decode bincode").0
```

Could you avoid using expect here and do some error handling?

---

What do you think would be the best approach here ? Should we return an error or default value instead of expect/panic ?

---

I think an error would be nice